doc/api/modules.md

```diff
@@ -197,13 +197,14 @@ LOAD_NODE_MODULES(X, START)
 NODE_MODULES_PATHS(START)
 1. let PARTS = path split(START)
 2. let I = count of PARTS - 1
-3. let DIRS = [GLOBAL_FOLDERS]
+3. let DIRS = []
 4. while I >= 0,
    a. if PARTS[I] = "node_modules" CONTINUE
    b. DIR = path join(PARTS[0 .. I] + "node_modules")
    c. DIRS = DIRS + DIR
```

Comment:
Just looked at this quickly, but on first read I think `DIRS` is entirely in the reverse ordering, since this loop is looping over from the shortest paths to the longest.

So a much simpler reodering might just be:

```suggestion
   c. DIRS = DIR + DIRS
```

but it would be worth checking the ordering isn't already specified as being this way around in the matching itself.

Reply:
@guybedford Thanks for reviewing.

If I understand correctly, the resolving ordering starts from the longest path to the shortest, (of course, plus `node_modules`). I think the loop in algorithm matches the behavior. As `I` is starts from `PARTS.length - 1` (longest) then down to `0` (shortest).

Reply:
Yes it seems I skimmed to quickly :) Reversing this loop could still be an alternative approach though.

Reply:
@Ayase-252 @guybedford  [source code](https://github.com/nodejs/node/blob/e46c680bf2b211bbd52cf959ca17ee98c7f657f5/lib/internal/modules/cjs/loader.js#L640) use `push`, but doc here seems `unshift`. Should it match algorithm in source code? 😂